crates/astria-composer/src/composer.rs

```diff
@@ -232,26 +311,144 @@ impl Composer {
                     &rollups,
                     rollup,
                     collector_exit,
+                    shutdown_token.clone()
                 );
             });
+        };
+
+        shutdown_info.run().await
+    }
+}
+
+struct ShutdownInfo {
+    api_server_shutdown_token: CancellationToken,
+    composer_shutdown_token: CancellationToken,
+    api_server_task_handle: Option<JoinHandle<eyre::Result<()>>>,
+    executor_task_handle: Option<JoinHandle<eyre::Result<()>>>,
+    grpc_server_task_handle: Option<JoinHandle<eyre::Result<()>>>,
+    geth_collector_tasks: JoinMap<String, eyre::Result<()>>,
+}
+
+impl ShutdownInfo {
+    async fn run(self) -> eyre::Result<()> {
+        let Self {
+            composer_shutdown_token,
+            api_server_shutdown_token,
+            api_server_task_handle,
+            executor_task_handle,
+            grpc_server_task_handle,
+            mut geth_collector_tasks,
+        } = self;
+
+        // if the composer is shutting down because of an unexpected shutdown from any one of the
+        // components(and not because of a SIGTERM), we need to send the cancel signal to all
+        // the other components.
+        composer_shutdown_token.cancel();
```

Comment:
this seems like the only spot where the cancel signal is sent.
this is fine for the collectors, but for executor it will only return to the main composer loop and trigger this shutdown after the executor's shutdown logic finished (seq actions and bundles all drained etc).
not sure if this is intended but figured i'd flag it here

Reply:
its a good catch. I don't think its a critical issue because the executor would exit mainly due to failures while submitting to sequencer. The shutdown logic would anyways fail while bundle submission. so it wouldn't cause any problems but yeah its not desirable and we would have unecessary logs.